app/models/job_invocation.rb

```diff
@@ -18,4 +18,28 @@ class JobInvocation < ActiveRecord::Base
   def to_action_input
     { :id => id, :name => job_name }
   end
+
+  def template_invocations_tasks
+    if last_task.present?
+      last_task.sub_tasks.for_action_types('Actions::RemoteExecution::RunHostJob')
+    else
+      ForemanTasks::Task.for_action_types('Actions::RemoteExecution::RunHostJob').where('1=0')
+    end
+  end
+
+  def failed_template_invocation_tasks
+    template_invocations_tasks.where(:result => 'warning')
+  end
+
+  def failed_host_ids
```

Comment:
exactly there it was planned to be used :-) note that `failed_hosts` is not used anywhere on purpose